### PR TITLE
front: fix yarn generate-types and add ci fail check

### LINKED ISSUE #6951: Fail CI step if yarn generate-types throw an error
### Who would benefit from this feature?

Developers and OSRD

### What is this feature about?

- If there is an error in the 'yarn generate-types' script, the CI doesn't fail; it generates wrong types in the frontend.

### Why is this feature valuable?

This feature matters because it:
It will prevent error in the frontend and desynchronization between openapi and front types

### Anything else people should know?

_No response_

### PR DESCRIPTION
**Pull request by Wadjetz**
close #6951

- [ ] For ProjectPathParams it a definition of the params that will be used to generate openapi, but it not used for now (work in progress) and it generate wrong typings

### CODE REVIEW

File: editoast/src/views/v2/train_schedule.rs [hunk 1]
```diff
@@ -411,7 +412,7 @@ pub struct SignalUpdate {
 /// Params are the infra_id and a list of train_ids
 #[utoipa::path(
     tag = "train_schedulev2",
-    params(ProjectPathParams),
+    // params(ProjectPathParams),
```
Comment (woshilapin) [inline editoast/src/views/v2/train_schedule.rs]: I'd say this comment can go away. Git history will remember it anyway.

File: editoast/src/views/v2/train_schedule.rs [hunk 2]
```diff
@@ -451,8 +452,7 @@ enum SimulationSummaryResultResponse {
 /// Useful for finding out whether pathfinding/simulation was successful.
 #[utoipa::path(
     tag = "train_schedulev2",
-    params(ProjectPathParams,
-    ("infra_id" = i64, Path, description = "The infra id")),
+    // params(ProjectPathParams),
```
Comment (woshilapin) [inline editoast/src/views/v2/train_schedule.rs]: I'd say this comment can go away. Git history will remember it anyway.

**Review by woshilapin — APPROVED** [review]
Nice fix about the CI. Although, it looks like this commit is doing two different things at once: fixing the CI and modifying the API. It would have been nice to split them (or make 2 different PR).

I don't understand why the `ProjectPathParams` are removed, could you add an explanation in the description of the PR so we can better track information about what happened and why?

File: editoast/src/views/v2/train_schedule.rs (hunk 1 above)
Comment (younesschrifi) [inline editoast/src/views/v2/train_schedule.rs]: You can remove this line

File: editoast/src/views/v2/train_schedule.rs (hunk 2 above)
Comment (younesschrifi) [inline editoast/src/views/v2/train_schedule.rs]: You can remove this line

**Review by younesschrifi — APPROVED** [review]
Nice PR :) 
Some things to change !

**Comment by Wadjetz** [comment]
For ProjectPathParams it a definition of the params that will be used to generate openapi, but it not used for now (work in progress) and it generate wrong typings

For two PR: I can't merge the CI fix without fixing the existing errors, so they are dependent on each others.

**Review by woshilapin — APPROVED** [review]

File: front/scripts/generate-types.sh
Comment (clarani) [inline front/scripts/generate-types.sh:1]: Thank you so much for this check :pray:

**Review by clarani — APPROVED** [review]
LGTM :white_check_mark:

**Review by Tguisnet — APPROVED** [review]

**Pull request merged**